### PR TITLE
implement threshold index api

### PR DESCRIPTION
**Pull request by nhsiehgit**
We need an index api so we can fetch all thresholds given a list of projects and/or environments

### CODE REVIEW

File: src/sentry/api/endpoints/release_thresholds/release_threshold_index.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.api.serializers import serialize
+from sentry.models.organization import Organization
+from sentry.models.release_threshold.release_threshold import ReleaseThreshold
+
+
+class ReleaseThresholdIndexGETSerializer(serializers.Serializer):
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/release_thresholds/release_threshold_index.py:16]: we tend to call input serializers `Validators`. know we could standardize this naming better

File: src/sentry/api/endpoints/release_thresholds/release_threshold_index.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                environment__in=environments_list,
+            )
+        if projects_list:
+            release_query &= Q(
+                project__in=projects_list,
+            )
+
+        release_thresholds = ReleaseThreshold.objects.filter(release_query)
+
+        return Response(serialize(list(release_thresholds), request.user), status=200)
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/release_thresholds/release_threshold_index.py:54]: do you need to worry about paginating?

Comment (nhsiehgit) [reply src/sentry/api/endpoints/release_thresholds/release_threshold_index.py:54]: pagination would probably be useful here.
will add that in

File: src/sentry/api/endpoints/release_thresholds/release_threshold_index.py (hunk 1 above)
Comment (nhsiehgit) [reply src/sentry/api/endpoints/release_thresholds/release_threshold_index.py:16]: cool - i can update the name

File: src/sentry/api/endpoints/release_thresholds/release_threshold_index.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    project = serializers.ListField(
+        required=True, allow_empty=False, child=serializers.IntegerField()
+    )
+
+
+@region_silo_endpoint
+class ReleaseThresholdIndexEndpoint(OrganizationEndpoint):
+    owner: ApiOwner = ApiOwner.ENTERPRISE
+    publish_status = {
+        "GET": ApiPublishStatus.PRIVATE,
```
Comment (cathteng) [inline src/sentry/api/endpoints/release_thresholds/release_threshold_index.py:30]: if we eventually want to publish this it should be EXPERIMENTAL

**Review by cathteng — APPROVED** [review]
nonblocking q: do you want to add tests for querying for a project (not the all projects indicator)?

File: src/sentry/api/endpoints/release_thresholds/release_threshold_index.py (hunk 3 above)
Comment (nhsiehgit) [reply src/sentry/api/endpoints/release_thresholds/release_threshold_index.py:30]: awesome will update

**Pull request merged**